File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        Arrays.sort(phone_book);

        for (int i=0; i<phone_book.length - 1; i++) {
            String lWord = phone_book[i];
            String rWord = phone_book[i+1];
            int len = Math.min(lWord.length(), rWord.length());
            boolean isDiff = false;
            for (int idx=0; idx<len; idx++) {
                if (lWord.charAt(idx) != rWord.charAt(idx)) {
                    isDiff = true;
                    break;
                }
            }
            if (!isDiff) return false;
        }
        return true;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()
    for i in range(len(phone_book)):
        if " " in phone_book[i]:
            phone_book[i] = num.split().join()
    
    
    for i in range(len(phone_book) - 1):
        answer = True
        f1 = phone_book[i]
        f2 = phone_book[i+1]
        min_l = min(len(f1), len(f2))
        isPass = True
        for j in range(min_l):
            if f1[j] != f2[j]:
                isPass = False
                break
        if isPass:
            return False
                
    return True